nwb_conversion_tools/datainterfaces/behavior/movie/moviedatainterface.py

```diff
@@ -203,37 +204,52 @@ def data_generator(file, count_max):
                         iter_axis=0,  # nwb standard is time as zero axis
                         maxshape=tuple(maxshape),
                     )
-                    image_series_kwargs.update(data=H5DataIO(mov, compression="gzip", chunks=best_gzip_chunk))
+                    data = H5DataIO(
+                        iterable,
+                        compression=compression,
+                        compression_options=compression_options,
+                        chunks=best_gzip_chunk,
+                    )
+                    data = H5DataIO(
+                        iterable, compression=compression, compression_opts=compression_options, chunks=best_gzip_chunk
+                    )
                 else:
-                    cap = cv2.VideoCapture(str(file))
-                    mov = []
-                    with tqdm(
-                        desc=f"Reading movie data for {Path(file).name}",
-                        position=tqdm_pos,
-                        total=total_frames,
-                        mininterval=tqdm_mininterval,
-                    ) as pbar:
-                        for _ in range(min(count_max, total_frames)):
-                            success, frame = cap.read()
-                            mov.append(frame)
-                            pbar.update(1)
-                    cap.release()
-                    image_series_kwargs.update(
-                        data=H5DataIO(
-                            DataChunkIterator(
-                                tqdm(
-                                    iterable=np.array(mov),
-                                    desc=f"Writing movie data for {Path(file).name}",
-                                    position=tqdm_pos,
-                                    mininterval=tqdm_mininterval,
-                                ),
-                                iter_axis=0,  # nwb standard is time as zero axis
-                                maxshape=tuple(maxshape),
+                    iterable = []
```

Comment:
```suggestion
                    iterable = np.zeros(shape=maxshape, dtype="uint8")
```
For better performance (mainly to avoid allocating the memory twice), let's just allocate a single array and fill its values iteratively.